[26기 장연정] Add: 메인페이지 기능 구현

yeonjeongj opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [x] 리뷰 반영
- [x] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- 홈페이지 내 콘텐츠들을 백엔드와의 소통을 통해 특정한 기준에 따라 다른 UI로 노출합니다.

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
1. 도현님 피드백 반영하여 코드 수정
- API 공통 관리를 위한 config.js 파일 추가

2. 데이터 통신을 위한 fetch 함수 구현 및 Routes.js 수정

<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
- 백엔드와의 커뮤니케이션을 통해 데이터를 주고받을 수 있게 되었습니다 :)

<br />

## :: 기타 질문 및 특이 사항

Commit:
Add: 메인 페이지 UI 구현
- 헤더 (문구 부분) 구현
- 키워드 섹션 목데이터 활용하여 UI 구현
- 작가 섹션 목데이터 활용하여 기본 UI 구현

we-dohyeon commented on src/pages/Main/Main.js:
```diff
@@ -1,9 +1,70 @@
+  }
+
+  handleKeywordsData = () => {
+    fetch('data/main/keywordsData.json', {
+      method: 'GET',
+    })
```
fetch함수의 `get method는 method의 default` 값입니다. 따라서 `생략이 가능`합니다. 
고로 get method 요청시에는 아래와 같이 두번째 인자가 생략 가능합니다. 

```jsx
const foo = () => {
    fetch(`${cartAPI}/quantity`)
    .then((res) => res.json())
```

yeonjeongj replied:
네 :-) method GET 이용 시에는 생략하도록 할게요!

we-dohyeon commented on src/pages/Main/Main.js:
```diff
@@ -1,9 +1,70 @@
+      );
+  };
+
+  componentDidMount() {
+    this.handleKeywordsData();
+    this.handleUsersData();
+  }
```
함수 분리 너무 좋네요!!!

we-dohyeon reviewed (commented):
연정님! 셀프리팩토링까지 정말 완벽하게 진행해주셨네요!!
대박입니다!!! 간단한 리뷰 한 가지 있고 추가적인 리뷰는 없습니다!
LGTM 👍

Commit:
Add: Section 1(LikedContents.js) 슬라이더 구현 완료
- 좌우 버튼 및 숫자 버튼 클릭 시 원하는 슬라이드로 이동하는 기능 구현 완료

Add: Section 1에 포함되는 SlideContent 컴포넌트

Add: Section 3 (Author.js) 온클릭 이벤트 구현 완료
- 클릭한 키워드 버튼만 색상 변하도록 기능 구현
- 작가 카드 내의 키워드 버튼 클릭 시 연결된 url로 이동하는 기능 구현

Commit:
Modify: 잘못 작성된 코드 수정

Commit:
Add : 컴포넌트 폴더 구조 변경
- 하나로 관리되던 컴포넌트 폴더를 각 섹션별로 구분하여 유지보수가 수월하도록 수정

Add : Section 1 (LikedContents) 내 마우스 hover 시 애니메이션 추가 구현
- 마우스 hover 시 이미지 줌인 & cover 색상 변경 구현

Commit:
Merge branch 'master' into feature/main

we-dohyeon commented on src/pages/Main/Main.js:
```diff
@@ -1,9 +1,126 @@
+      <div className="main">
+        {/* <Nav /> */}
+        {/* <Header /> */}
+        <header>
+          <h1>맛을 나누는 공간, 아침엔 브런치</h1>
+          <p>나를 사로잡은 매력적인 브런치를 소개해 보세요.</p>
+          <p> 그리고 다시 느껴보세요.</p>
+          <p> 세상 모든 곳의 즐거움을.</p>
+        </header>
```
header 태그는 한 페이지에 한 번 사용해주세요!
해당 페이지에서는 없지만 결국 main에는 Header, Nav 컴포넌트가 들어오게되니 중복이되겠죠?!

yeonjeongj replied:
넵 확인했습니다 :)

we-dohyeon commented on src/pages/Main/LikedContents/LikedContents.js:
```diff
@@ -0,0 +1,176 @@
+            <i class="fas fa-chevron-right" />
+          </button>
+        </div>
+        <ul className="contentsOrder">
+          <li
+            value="1"
+            onClick={this.moveToSlide}
+            className={activeBtn === '1' ? 'numBtn active' : 'numBtn'}
+          >
+            01
+          </li>
+          <li
+            value="2"
+            onClick={this.moveToSlide}
+            className={activeBtn === '2' ? 'numBtn active' : 'numBtn'}
+          >
+            02
+          </li>
+          <li
+            value="3"
+            onClick={this.moveToSlide}
+            className={activeBtn === '3' ? 'numBtn active' : 'numBtn'}
+          >
+            03
+          </li>
+          <li
+            value="4"
+            onClick={this.moveToSlide}
+            className={activeBtn === '4' ? 'numBtn active' : 'numBtn'}
+          >
+            04
+          </li>
+        </ul>
```
해당 부분 [1,2,3,4] 배열을 활용해서 중복을 줄여주세요!!

yeonjeongj replied:
네 수정하였습니다!

we-dohyeon reviewed (commented):
연정님! 구두 리뷰 포함해서 리팩토링 진행해주세요!!!

Commit:
Refactor: 멘토님 리뷰 반영하여 전반적인 리팩토링 진행
- Main.js <header> 태그 변경
- LikedContent.js와 Authors.js 파일에서 map() 메소드 활용하여 중복코드 삭제
- 불필요한 코드 삭제

Commit:
Add: 백엔드와의 데이터 통신을 위한 fetch 함수 구현

Commit:
Add: local master branch의 업데이트 부분 merge

Merge branch 'master' into feature/main

Commit:
Add: 머지된 컴포넌트 반영에 따른 디자인 수정

Commit:
Modify: 머지에 따른 scss 수정

Add: API 공통 관리를 위한 config.js 추가

Add: path parameter 사용을 위한 Routes.js 수정

we-dohyeon reviewed (commented):
LGTM 👍
